packages/client/src/links/httpLink.ts

```diff
@@ -20,13 +21,22 @@ export function httpLink<TRouter extends AnyRouter>(
         });
         promise
           .then((res) => {
-            observer.next({
+            const result = {
               context: res.meta,
-              data: res.json as any,
-            });
+              data: res.json,
+            } as OperationResult<TRouter, any>;
+            const transformed = transformOperationResult(result, runtime);
+            if (transformed.ok) {
+              observer.next({
+                context: res.meta,
+                data: res.json as any,
```

Comment:
```suggestion
                data: res.data as any,
```

This should probably return `transformed.data` but that requires a bunch of refactoring which I think is a symptom of something larger we should fix before we have a public beta. I have previously written a note about revisiting observables a bit closer. 

Currently, we have this 

```ts
type OperationResultData<TRouter extends AnyRouter, TOutput> =
  | TRPCResponse<TOutput, inferRouterError<TRouter>>
  | TRPCResponseMessage<TOutput, inferRouterError<TRouter>>;

export interface OperationResult<TRouter extends AnyRouter, TOutput> {
  data: OperationResultData<TRouter, TOutput>;
  context?: OperationContext;
}
```

& I think `data` should only be the success state here.


I don't have any time to code in the next few days as I'm traveling & going to a wedding tomorrow, but maybe @sachinraja or @jlalmes can help out

This is a bit of a remain from how the behavior was previous to #1855 and we missed doing this refactor.

Reply:
Regarding the suggested change, there is no `res.data`, only `res.json` and `res.meta`. Also, I had the same concern as you in the Discord thread, thanks for mentioning it here as well. And I agree that having consistency of returning transformed data would be better here. I think that will be a bit too much for me to tackle, so help is appreciated.

Reply:
Yeah, I think only me, @sachinraja, or @jlalmes are the right ppl for this job 😅